plugin/audit_log/buffer.c

```diff
@@ -163,11 +167,25 @@ void audit_log_buffer_shutdown(audit_log_buffer_t *log)
 int audit_log_buffer_write(audit_log_buffer_t *log, const char *buf, size_t len)
 {
   if (len > log->size)
-    return(1);
+  {
+    if (!log->drop_if_full)
+    {
+      mysql_mutex_lock(&log->mutex);
```

Comment:
If one runs with big queries and too small log buffer, they will be in this mode constantly even though configuration will say "async", with performance surprises. Or maybe user could bump the log buffer a bit and it will fit its queries, but he will never learn this. Does it make sense to diagnose this somehow? Not in the scope of this PR